src/main/java/org/opensearch/neuralsearch/processor/factory/SparseEncodingProcessorFactory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor.factory;
+
+import static org.opensearch.ingest.ConfigurationUtils.readMap;
+import static org.opensearch.ingest.ConfigurationUtils.readStringProperty;
+import static org.opensearch.neuralsearch.processor.TextEmbeddingProcessor.*;
+
+import java.util.Map;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.env.Environment;
+import org.opensearch.ingest.Processor;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+import org.opensearch.neuralsearch.processor.SparseEncodingProcessor;
+
+@Log4j2
+public class SparseEncodingProcessorFactory implements Processor.Factory {
```

Comment:
Please add description of the class

Reply:
Sure, also add one for TextEmbeddingProcessorFactory